crates/bevy_core_pipeline/src/upscaling/mod.rs

```diff
@@ -0,0 +1,119 @@
+mod node;
+
+pub use node::UpscalingNode;
+
+use bevy_app::prelude::*;
+use bevy_asset::{Assets, HandleUntyped};
+use bevy_ecs::prelude::*;
+use bevy_render::renderer::RenderDevice;
+use bevy_render::texture::BevyDefault;
+use bevy_render::view::ExtractedView;
+use bevy_render::{render_resource::*, RenderApp, RenderStage};
+
+use bevy_reflect::TypeUuid;
+
+const UPSCALING_SHADER_HANDLE: HandleUntyped =
+    HandleUntyped::weak_from_u64(Shader::TYPE_UUID, 14589267395627146578);
+
+pub struct UpscalingPlugin;
+
+impl Plugin for UpscalingPlugin {
+    fn build(&self, app: &mut App) {
+        let mut shaders = app.world.get_resource_mut::<Assets<Shader>>().unwrap();
+        shaders.set_untracked(
+            UPSCALING_SHADER_HANDLE,
+            Shader::from_wgsl(include_str!("upscaling.wgsl")),
+        );
+
+        app.sub_app_mut(RenderApp)
+            .init_resource::<UpscalingPipeline>()
+            .init_resource::<SpecializedPipelines<UpscalingPipeline>>()
+            .add_system_to_stage(RenderStage::Queue, queue_upscaling_bind_groups);
+    }
+}
+
+pub struct UpscalingPipeline {
+    ldr_texture_bind_group: BindGroupLayout,
+}
+
+impl FromWorld for UpscalingPipeline {
+    fn from_world(render_world: &mut World) -> Self {
+        let render_device = render_world.get_resource::<RenderDevice>().unwrap();
+
+        let hdr_texture_bind_group =
+            render_device.create_bind_group_layout(&BindGroupLayoutDescriptor {
+                label: Some("upscaling_ldr_texture_bind_group_layout"),
+                entries: &[
+                    BindGroupLayoutEntry {
+                        binding: 0,
+                        visibility: ShaderStages::FRAGMENT,
+                        ty: BindingType::Texture {
+                            sample_type: TextureSampleType::Float { filterable: false },
+                            view_dimension: TextureViewDimension::D2,
+                            multisampled: false,
+                        },
+                        count: None,
+                    },
+                    BindGroupLayoutEntry {
+                        binding: 1,
+                        visibility: ShaderStages::FRAGMENT,
+                        ty: BindingType::Sampler(SamplerBindingType::NonFiltering),
```

Comment:
I feel like this should at least be an option like nearest-neighbour / bilinear. Then later we can add AMD FSR, NVIDIA DLSS, Intel's offering, something TAA-based as part of the TAA node perhaps, or whatever else we like.

Random curiosity that I haven't thought about before: is non-filtering practically the same as filtering with `FilterMode::Nearest`?

Reply:
I added a `UpscalingPipelineKey` with support for specifying the mode, but it's not exposed in a usable way yet.

Hdr and the upscaling mode should probably be configuration options on the render target? I'll wait until the new render target changes are finalized.

Reply:
I guess two bind groups would be needed, one filtering and one not.